ddl/ddl_api.go

```diff
@@ -1161,6 +1163,61 @@ func (d *ddl) AddColumn(ctx sessionctx.Context, ti ast.Ident, spec *ast.AlterTab
 	return errors.Trace(err)
 }
 
+func (d *ddl) AddTablePartitions(ctx sessionctx.Context, ti ast.Ident, spec *ast.AlterTableSpec) error {
+	is := d.infoHandle.Get()
+	schema, ok := is.SchemaByName(ti.Schema)
+	if !ok {
+		return errors.Trace(infoschema.ErrDatabaseNotExists.GenByArgs(schema))
+	}
+	t, err := is.TableByName(ti.Schema, ti.Name)
+	if err != nil {
+		return errors.Trace(infoschema.ErrTableExists.GenByArgs(ti.Schema, ti.Name))
+	}
+	if len(spec.PartDefinitions) == 0 {
+		return errors.Trace(infoschema.ErrPartitionsMustBeDefined)
+	}
+
+	partitionInfo := &model.PartitionInfo{}
+	if t.Meta().Partition != nil {
+		partitionInfo = &model.PartitionInfo{
+			Type: t.Meta().Partition.Type,
+			Expr: t.Meta().Partition.Expr,
+		}
+		for _, def := range spec.PartDefinitions {
+			pid, err1 := d.genGlobalID()
+			if err1 != nil {
+				return errors.Trace(err1)
+			}
+			piDef := model.PartitionDefinition{
+				Name:    def.Name,
+				ID:      pid,
+				Comment: def.Comment,
+			}
+			for _, expr := range def.LessThan {
+				buf := new(bytes.Buffer)
```

Comment:
piDef.LessThan = append(piDef, expr) is ok?

Reply:
Cannot use expr as type string in append.